### ISSUE TITLE
Filename too long

### ISSUE THREAD

**Issue by AlexanderThomme** [AthenaDecisionSystems/athena-owl-demos#7]
Systeminfo:
OS Name:                   Microsoft Windows 11 Famille
OS Version:                10.0.22631 N/A Build 22631

From C:\home\athomme\work\athena-decision-systems\dev\athena-owl-agent, which is a reasonably short path I am running:
`git clone https://github.com/AthenaDecisionSystems/athena-owl-demos.git`

As a result, I am getting:
error: unable to create file IBU-insurance-demo/decisions/workspace/rules-insurance-pc-claims-nba/rules/Complaint Handling/Applied coverages/Targeted/Auto/AC-AUTO-SUBV-DISC - Discount on substitution vehicle.brl: Filename too long
error: unable to create file IBU-insurance-demo/decisions/workspace/rules-insurance-pc-claims-nba/rules/Complaint Handling/Applied coverages/Targeted/Auto/AC-AUTO-SUBV-UP - Upgrade policy with substitution vehicle.brl: Filename too long
error: unable to create file IBU-insurance-demo/decisions/workspace/rules-insurance-pc-claims-nba/rules/Complaint Handling/Applied coverages/Targeted/Auto/AC-AUTO-TPL - 3rd party policy and responsible for damage.dta: Filename too long
error: unable to create file IBU-insurance-demo/decisions/workspace/rules-insurance-pc-claims-nba/rules/Complaint Handling/Applied coverages/Targeted/Home/AC-HOME-CONT-UP - Upgrade policy to include content.brl: Filename too long
error: unable to create file IBU-insurance-demo/decisions/workspace/rules-insurance-pc-claims-nba/rules/Complaint Handling/Applied coverages/Targeted/Home/AC-HOME-CONT-VOUCHER - Propose voucher for extra service with affiliate provider.brl: Filename too long
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.

**Comment by jcgardner** [comment]
This is due to Windows limit imposed on  the maximum path length of files. Path length includes the full folder path and the file name and it is limited to 260 characters. The IBU insurance ODM project was created on a Mac where such a limit does not exist.
Possible solution is just to rename rule packages and rule names to make sure all path lengths are well under 260.

**Comment by AlexanderThomme** [comment]
Exactly, this is a known issue with Windows/DOS. According to google the limit is 256.
Of course the problem will always happen if the repository is cloned in a directory deep in the file system, but we should avoid that risk and 210 character of 256 is too much.
I see no other solution that the one you propose. Best is to use git-mv rather than mv for tracking purpose.

**Comment by stockcrack** [comment]
We should probably shorten the name of the demo as well… IBU-insurance-demo is already a lot to type.

--
Harley


From: AlexanderThomme ***@***.***>
Date: Friday, 26 July 2024 at 16:18
To: AthenaDecisionSystems/athena-owl-demos ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [AthenaDecisionSystems/athena-owl-demos] Filename too long (Issue #7)

Exactly, this is a known issue with Windows/DOS. According to google the limit is 256.
Of course the problem will always happen if the repository is cloned in a directory deep in the file system, but we should avoid that risk and 210 character of 256 is too much.
I see no other solution that the one you propose. Best is to use git-mv rather than mv for tracking purpose.

**Comment by AthenaDecisionSystems** [comment]
and if we could remove space in folder and file name it will be great too. we can remove -demo as we are in a demo repo anyway.

________________________________
